src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/__init__.py

```diff
@@ -0,0 +1,111 @@
+               ' $ az vm create --help\n'\
+               ' $ az feedback\n',
+}
+
+def _prompt_global_output(global_config):
+    try:
+        config_output = global_config.get('core', 'output')
+        default_output = [i for i, x in enumerate(OUTPUT_LIST) if x['name'] == config_output][0]+1
+    except (IndexError, configparser.NoSectionError, configparser.NoOptionError):
+        default_output = 1
+    options = '\n'.join([' [{}] {} - {}'.format(i+1, x['name'], x['desc']) \
+                        for i, x in enumerate(OUTPUT_LIST)])
+    allowed_vals = list(range(1, len(OUTPUT_LIST)+1))
+    while True:
+        try:
+            ans = int(input(MESSAGES['prompt_global_output'].format(options, default_output)) \
+                            or default_output)
+            if ans in allowed_vals:
+                # array index is 0-based, user input is 1-based
+                return ans-1
+            raise ValueError
+        except ValueError:
+            logger.warning('Valid values are %s', allowed_vals)
+
+def _prompt_global_settings_exist():
+    while True:
+        ans = input(MESSAGES['global_settings_exists'])
+        if not ans or ans.lower() == 'n':
+            return False
+        if ans.lower() == 'y':
+            return True
+
+def _print_cur_configuration(file_config):
+    for section in file_config.sections():
+        print('[{}]'.format(section))
+        for option in file_config.options(section):
+            print('{} = {}'.format(option, file_config.get(section, option)))
+        print()
+    env_vars = [ev for ev in os.environ if ev.startswith(ENV_VAR_PREFIX)]
+    if env_vars:
+        print(MESSAGES['env_vars_heading'])
+        print('\n'.join(['{} = {}'.format(ev, os.environ[ev]) for ev in env_vars]))
+        print()
+
+def _handle_global_configuration():
+    print(MESSAGES['global_settings_loc'].format(GLOBAL_CONFIG_PATH))
+    print(MESSAGES['active_env_settings_loc'].format(ACTIVE_ENV_CONFIG_PATH))
+    # We don't want to use AzConfig as we don't want to use get values set by env vars.
+    file_config = configparser.SafeConfigParser()
+    config_exists = file_config.read([GLOBAL_CONFIG_PATH, ACTIVE_ENV_CONFIG_PATH])
+    global_config = configparser.SafeConfigParser()
+    global_config.read(GLOBAL_CONFIG_PATH)
+    change_global_settings = None
+    if config_exists:
+        print(MESSAGES['current_config_info'])
+        _print_cur_configuration(file_config)
+        change_global_settings = _prompt_global_settings_exist()
+    if not config_exists or change_global_settings:
+        output_index = _prompt_global_output(global_config)
+        try:
+            global_config.add_section('core')
+        except configparser.DuplicateSectionError:
+            pass
+        global_config.set('core', 'output', OUTPUT_LIST[output_index]['name'])
+        with open(GLOBAL_CONFIG_PATH, 'wb') as configfile:
+            global_config.write(configfile)
+
+def handle_configure():
```

Comment:
I will point out that adding it to the PR description makes it hard to make comments on it. It's a little easier if you post such things as a comment so any subsequent comments follow naturally. 

---
In reply to: [76337995](https://github.com/Azure/azure-cli/pull/739#discussion_r76337995) [](ancestors = 76337995)